core/frontend/src/test/Viewport.test.ts

```diff
@@ -0,0 +1,116 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { expect } from "chai";
+import { ScreenViewport } from "../Viewport";
+import { Marker } from "../Marker";
+import { Decorator } from "../ViewManager";
+import { Point3d } from "@bentley/geometry-core";
+import { DecorateContext } from "../ViewContext";
+import { IModelApp } from "../IModelApp";
+import { SpatialViewState } from "../SpatialViewState";
+// import { NoRenderApp } from "../NoRenderApp";
+import { IModelConnection } from "../IModelConnection";
+
+describe("ScreenViewport", () => {
+  beforeEach(async () => {
+    await IModelApp.startup();
+  });
+  afterEach(async () => {
+    if (IModelApp.initialized) await IModelApp.shutdown();
+  });
+
+  function makeMarker(vp: ScreenViewport) {
+    const marker = new Marker(vp.viewToWorld({x: 0, y: 0, z: 0}), { x: 10, y: 10 });
+    marker.htmlElement = document.createElement("div");
+    return marker;
+  }
+
+  class AddMarkersAlwaysDecorator implements Decorator {
+    public markers: Marker[];
+    public constructor(vp: ScreenViewport) {
+      this.markers = new Array(3).fill(undefined).map(() => makeMarker(vp));
+    }
+    public decorate(ctx: DecorateContext) {
+      this.markers.forEach((m) => {
+        m.addDecoration(ctx);
+      });
+    }
+  }
+
+  class AddMarkersOnceDecorator implements Decorator {
+    public markers: Marker[];
+    private _didOnce = false;
+    public constructor(vp: ScreenViewport) {
+      this.markers = new Array(3).fill(undefined).map(() => makeMarker(vp));
+    }
+    public decorate(ctx: DecorateContext) {
+      if (!this._didOnce) {
+        this.markers.forEach((m) => {
+          m.addDecoration(ctx);
+        });
+        this._didOnce = true;
+      }
+    }
+  }
+
+  function createMockViewport() {
+    const parentDiv = document.createElement("div");
+    parentDiv.setAttribute("height", "100px");
+    parentDiv.setAttribute("width", "100px");
+    parentDiv.style.height = parentDiv.style.width = "100px";
+    document.body.appendChild(parentDiv);
+    const vp = ScreenViewport.create(
+      parentDiv,
+      SpatialViewState.createBlank(
+        {} as IModelConnection,
```

Comment:
Use a `BlankConnection` instead?

Reply:
I wasn't sure if that was appropriate when reading it, will try

Reply:
[Here's an example](https://github.com/imodeljs/imodeljs/blob/b9d44e26b6fbffb256eae880c691d2aec04cce97/core/frontend/src/test/render/webgl/SurfaceTransparency.test.ts#L66).